src/modules/tools/endstops/Endstops.cpp

```diff
@@ -376,12 +382,11 @@ void Endstops::move_to_origin(std::bitset<3> axis)
     // float pos[3]; THEROBOT->get_axis_position(pos); if(pos[0] == 0 && pos[1] == 0) return;
 
     this->status = MOVE_TO_ORIGIN;
-    // Move to center using a regular move, use slower of X and Y fast rate
-    float rate = std::min(this->fast_rates[0], this->fast_rates[1]) * 60.0F;
+    // Move to center using a regular G0 move, use whatever feed rate was previously defined for G0
     char buf[32];
     THEROBOT->push_state();
     THEROBOT->inch_mode = false;     // needs to be in mm
-    snprintf(buf, sizeof(buf), "G53 G0 X0 Y0 F%1.4f", rate); // must use machine coordinates in case G92 or WCS is in effect
+    snprintf(buf, sizeof(buf), "G53 G0 X0 Y0"); // must use machine coordinates in case G92 or WCS is in effect
```

Comment:
I see what you mean and agree with you.  

My understanding of G0 is that it is always a rapid rate and not something that changes.  If you want to move something at a speed other than rapid G1 should be used.   In fact on all the old CNC controllers I have ever used, G0 had no option for an F parameter

Would it be more appropriate to just use default_seek_rate for this?  it would get put back to whatever was in there modal after the pop_state